tests/integration/test_charm.py

```diff
@@ -6,27 +6,204 @@
+                apps=[
+                    helpers.KAFKA_NAME,
+                    helpers.OPENSEARCH_NAME,
+                    helpers.POSTGRES_NAME,
+                    helpers.CERTIFICATES_NAME,
+                    helpers.ZOOKEPER_NAME,
+                ],
+                status="active",
+                timeout=15 * 60,
+            )
+
+            # Create offers for DataHub to consume
+            logger.info("Creating offers")
+            await asyncio.gather(
+                ops_test.model.create_offer(
+                    endpoint="kafka-client", application_name=helpers.KAFKA_NAME, offer_name=helpers.KAFKA_OFFER_NAME
+                ),
+                ops_test.model.create_offer(
+                    endpoint="opensearch-client",
+                    application_name=helpers.OPENSEARCH_NAME,
+                    offer_name=helpers.OPENSEARCH_OFFER_NAME,
+                ),
+                ops_test.model.create_offer(
+                    endpoint="database", application_name=helpers.POSTGRES_NAME, offer_name=helpers.POSTGRES_OFFER_NAME
+                ),
+            )
+
+        with ops_test.model_context(k8s_model):
+            async with ops_test.fast_forward(fast_interval="5m"):
+                # Deploy DataHub
+                logger.info("Deploying '%s'", helpers.APP_NAME)
+                await helpers.deploy_charm(ops_test, charm)
+
+                # TODO (mertalpt): Find a way to avoid doing this.
+                # Gives `pebble_ready` time to run before making relations.
+                # I suspect it is a problem because gigabytes of OCI images are not cached
+                # and take a long time to be downloaded.
+                await asyncio.sleep(10 * 60)
+
+                # Wait for DataHub to settle
+                logger.info("Waiting for '%s' to settle into 'blocked-idle'", helpers.APP_NAME)
+                await ops_test.model.wait_for_idle(
+                    apps=[helpers.APP_NAME],
+                    status="blocked",
+                    raise_on_blocked=False,
+                    timeout=10 * 60,
+                )
+
+                # Consume offers
+                logger.info("Consuming offers")
+                await ops_test.juju("consume", f"{lxd_model}.{helpers.KAFKA_OFFER_NAME}")
+                await ops_test.juju("consume", f"{lxd_model}.{helpers.OPENSEARCH_OFFER_NAME}")
+                await ops_test.juju("consume", f"{lxd_model}.{helpers.POSTGRES_OFFER_NAME}")
+
+                # Relate to offers
+                logger.info("Integrating to '%s'", helpers.KAFKA_OFFER_NAME)
+                await ops_test.model.integrate(helpers.APP_NAME, helpers.KAFKA_OFFER_NAME)
+                logger.info(
+                    "Waiting for '%s' to settle into 'blocked-idle' post '%s' integration",
+                    helpers.APP_NAME,
+                    helpers.KAFKA_OFFER_NAME,
+                )
+                await asyncio.sleep(2 * 60)
```

Comment:
same comment as above

Reply:
Similar to above, this can be replaced when status changes behave better after an overhaul.